code/modules/reagents/machinery/distillery.dm

```diff
@@ -53,31 +53,17 @@
 	var/image/overlay_dumping
 	var/image/overlay_connected
 
-// Our unique beaker, used in its unique recipes to ensure things can only react inside this machine and minimize oddities from trying to transfer to a machine and back.
-	var/obj/item/weapon/reagent_containers/glass/distilling/Reservoir
-
 	var/obj/item/weapon/reagent_containers/glass/InputBeaker
 	var/obj/item/weapon/reagent_containers/glass/OutputBeaker
 
 // A multiplier for the production amount. This should really only ever be lower than one, otherwise you end up with duping.
 	var/efficiency = 1
 
-/obj/item/weapon/reagent_containers/glass/distilling
-	name = "distilling chamber"
-	desc = "You should not be seeing this."
-	volume = 600
-
-	var/obj/machinery/portable_atmospherics/powered/reagent_distillery/Master
-
-/obj/item/weapon/reagent_containers/glass/distilling/Destroy()
-	Master = null
-	..()
-
 /obj/machinery/portable_atmospherics/powered/reagent_distillery/Initialize()
 	. = ..()
 
-	Reservoir = new (src)
-	Reservoir.Master = src
+	reagents = new /datum/reagents/distilling(600, src)
```

Comment:
Personally I would suggest allowing `create_reagents` to take a type argument for the new reagent instance rather than doing a naked `new`.